src/components/Profile.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import Search from './Search';
+import EachProfile from './EachProfile';
+import data from '../list.json';
+
+export default function Profile({ whichState }) {
+  const [firstName, setFirstName] = useState('');
+  const [searchResult, setSearchResult] = useState(data);
+  const [searching, setSearching] = useState(false);
+
+  const filterProfile = (find) => {
+    if (find === '') {
+      setSearchResult(data); //초기화면
+      setSearching(false);
+    } else {
+      //검색한거만 필터링
+      const result = data.filter((element) => element.name.indexOf(find) === 0);
+      setSearchResult(result);
+    }
```

Comment:
이런식으로 조건부 렌더링하셔도 좋을 것 같아요~
```suggestion
   find && (
      const result = data.filter((element) => element.name.indexOf(find) === 0);
      setSearchResult(result));
    }
```

Reply:
앗 감사합니다!!  참여못해서 죄송합니다 ㅠㅠ

프론트 여러분 보이시나요...?✋✋